### PR TITLE
Remove `kube-dns` from `dns-providers`

### PR DESCRIPTION
**Pull request by mateoflorido**
## Purpose
Kube-DNS is included in K8s Legacy addons. We should remove it now that deployment on 1.25+ does not work.
## Changes
- Remove `Kube DNS` from the valid dns providers.
- Modify config.yaml to reflect the new providers list.
- Update README to reflect the changes.

### CODE REVIEW

File: reactive/kubernetes_control_plane.py [hunk 1]
```diff
@@ -3366,8 +3356,6 @@ def get_dns_provider():
         if not dns_provider:
             if "core-dns" in valid_dns_providers:
                 dns_provider = "core-dns"
```
Comment (kwmonroe) [inline reactive/kubernetes_control_plane.py:3358]: small knit; this `if` isn't needed anymore.
```suggestion
            dns_provider = "core-dns"
```

**Review by kwmonroe — APPROVED** [review]
LGTM with a tiny suggestion.

**Comment by mateoflorido** [comment]
There are some issues with `setuptools` and `wheel` dependencies, affecting the build and run process of some charms in Focal (20.04). More information: [Chamhub](https://discourse.charmhub.io/t/cannot-install-dependencies-modulenotfounderror-no-module-named-setuptools-command-build/7374)

**Comment by mateoflorido** [comment]
It looks like the dependency issue has been sorted out. Unpinning the dependency.

**Review by addyess — APPROVED** [review]
LGTM2

**Pull request merged**